keyboards/charue/supernova/keymaps/debug/keymap.c

```diff
@@ -0,0 +1,75 @@
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+// Defines names for use in layer keycodes and the keymap
+enum layer_names {
+    _BASE,
+    _FN
+};
+
+// Defines the keycodes used by our macros in process_record_user
+enum custom_keycodes {
+    QMKBEST = SAFE_RANGE,
+    QMKURL
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    /* Base */
+    [_BASE] = LAYOUT_all(
+      RGB_TOG,         RGB_MOD, KC_F2, KC_F3, KC_F4,    KC_F5,  KC_F6,  KC_F7,  KC_F8,      KC_F9,   KC_F10, KC_F11,  KC_F12,     KC_PSCR, KC_SLCK, KC_PAUS,
+      KC_GRV,  KC_1,  KC_2,  KC_3,  KC_4,  KC_5,  KC_6,  KC_7,  KC_8,   KC_9,   KC_0,    KC_MINS, KC_EQL, KC_BSPC, MO(_FN),    KC_INS,  KC_HOME, KC_PGUP,
+       KC_TAB,   KC_Q,  KC_W,  KC_E,  KC_R,  KC_T,  KC_Y,  KC_U,  KC_I,   KC_O,   KC_P,    KC_LBRC, KC_RBRC,    KC_BSLS,       KC_DEL,  KC_END,  KC_PGDN,
+         KC_CAPS, KC_A,  KC_S,  KC_D,  KC_F,  KC_G,  KC_H,  KC_J,  KC_K,   KC_L,   KC_SCLN, KC_QUOT,       KC_ENT,
+  KC_LSFT, QMKURL,  KC_Z,  KC_X,  KC_C,  KC_V,  KC_B,  KC_N,  KC_M, KC_COMM, KC_DOT, KC_SLSH, KC_RSFT,  QMKBEST,                        KC_UP,
+      KC_LCTRL, KC_LGUI, KC_LALT, KC_LGUI,   KC_SPC,      KC_SPC,     KC_RGUI,    KC_RALT,   KC_RGUI,  KC_RCTRL,               KC_LEFT, KC_DOWN, KC_RIGHT
+    ),
+    [_FN] = LAYOUT_all(
+      RESET,          KC_F1, KC_F2, KC_F3, KC_F4,    KC_F5,  KC_F6,  KC_F7,  KC_F8,      KC_F9,   KC_F10, KC_F11,  KC_F12,     KC_PSCR, KC_SLCK, KC_PAUS,
+      KC_GRV,  KC_1,  KC_2,  KC_3,  KC_4,  KC_5,  KC_6,  KC_7,  KC_8,   KC_9,   KC_0,    KC_MINS, KC_EQL, KC_BSPC, KC_TRNS,     KC_INS,  KC_HOME, KC_PGUP,
+       KC_TAB,   KC_Q,  KC_W,  KC_E,  KC_R,  KC_T,  KC_Y,  KC_U,  KC_I,   KC_O,   KC_P,    KC_LBRC, KC_RBRC,    KC_BSLS,       KC_DEL,  KC_END,  KC_PGDN,
+         KC_CAPS, KC_A,  KC_S,  KC_D,  KC_F,  KC_G,  KC_H,  KC_J,  KC_K,   KC_L,   KC_SCLN, KC_QUOT,       KC_ENT,
+  KC_LSFT, QMKURL,  KC_Z,  KC_X,  KC_C,  KC_V,  KC_B,  KC_N,  KC_M, KC_COMM, KC_DOT, KC_SLSH, KC_RSFT,  QMKBEST,                        KC_UP,
+      KC_LCTRL, KC_LGUI, KC_LALT, KC_LGUI,   KC_SPC,      KC_SPC,     KC_RGUI,    KC_RALT,   KC_RGUI,  KC_RCTRL,               KC_LEFT, KC_DOWN, KC_RIGHT
+    )
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        case QMKBEST:
+            if (record->event.pressed) {
+                // when keycode QMKBEST is pressed
+                SEND_STRING("QMK is the best thing ever!");
+            } else {
+                // when keycode QMKBEST is released
+            }
+            break;
+        case QMKURL:
+            if (record->event.pressed) {
+                // when keycode QMKURL is pressed
+                SEND_STRING("https://qmk.fm/\n");
+            } else {
+                // when keycode QMKURL is released
+            }
+            break;
+    };
+    return true;
+};
```

Comment:
```suggestion
```

Reply:
I am aware that these are examples of macros. However, this is a debug keymap. It's important to check the macro functionality as well. They are not present in the default or via keymaps.